site/pages/concepts/middleware.mdx

```diff
@@ -0,0 +1,27 @@
+---
+title: Middleware
+description: What is Middleware?
+---
+
+# Middleware
+
+The [Smart Account Client](/concepts/smart-account-client) is extended with a series of a middleware. When sending or building user operations for signing and sending, the flow can be pretty involved.
+Sending a UO requires you to get the latest nonce for an account, check if the account is deployed to set the `initCode` or `factory` and `factoryData` in the user operation, estimate fees, estimate gas, and then sign the user operation.
+If you want to use a paymaster, you need to generate some `dummyPaymasterAndData` to use during gas estimation, and after estimating gas you can request gas sponsorship from your paymaster.
+Middleware allows you to avoid having to write the same flows over and over when all you want to do is call a contract with some data or send some ETH to a destination.
+
+## Middleware order
+
+:::tip
+When using the [React](/react/overview), [Core](/core/overview), or [Infra](/infra/overview) packages, the Smart Account Clients are already configured to use the appropriate middleware for your needs.
+If you want to sponsor gas, the Smart Account Client's in these packages abstract the relevant middleware away from you since all you need is a `policyId` to sponsore gas with Alchemy.
```

Comment:
```suggestion
If you want to sponsor gas, the Smart Account Clients in these packages abstract the relevant middleware away from you since all you need is a `policyId` to sponsor gas with Alchemy.
```